cmd/fetch.go

```diff
@@ -34,6 +37,7 @@ func NewFetchCmd(globalCfg *config.GlobalImpl) *cobra.Command {
 	f.IntVar(&fetchOptions.Concurrency, "concurrency", 0, "maximum number of concurrent helm processes to run, 0 is unlimited")
 	f.BoolVar(&fetchOptions.SkipDeps, "skip-deps", false, `skip running "helm repo update" and "helm dependency build"`)
 	f.StringVar(&fetchOptions.OutputDir, "output-dir", "", "directory to store charts (default: temporary directory which is deleted when the command terminates)")
+	f.StringVar(&fetchOptions.OutputDirTemplate, "output-dir-template", "", fmt.Sprintf("go text template for generating the output directory (example: \"%s\")", path.Join("{{ .OutputDir }}", "{{ .Release.Name }}")))
```

Comment:
I think we should add more info about this arg. so users can use this easily.

Reply:
and default value as well?

Reply:
> I think we should add more info about this flag. so users can use this easily.

I copied the description from the `--output-dir-template` flag of the `template` command. (And replaced the "Default" with an "Example".) I do not really know what to change here.

> and default value as well?

The default value will probably be a long one because it consist of up to 8 elements (depending on the release spec). But I can do that. 🙂

Reply:
This is what the default output-dir might look like as a template string:

```tpl
{{ .OutputDir }}{{ if .Release.TillerNamespace }}/{{ .Release.TillerNamespace }}{{ end }}{{ if .Release.Namespace }}/{{ .Release.Namespace }}{{ end }}{{ if .Release.KubeContext }}/{{ .Release.KubeContext }}{{ end }}/{{ .Release.Name }}/{{ .ChartName }}/{{ if .Release.Version }}{{ .Release.Version }}{{ else }}latest{{ end }}
```

Reply:
And this is what `helmfile fetch --help` would print:

> ```
> Fetch charts from state file
> 
> Usage:
>   helmfile fetch [flags]
> 
> Flags:
>       --concurrency int              maximum number of concurrent helm processes to run, 0 is unlimited
>       --output-dir string            directory to store charts (default: temporary directory which is deleted when the command terminates)
>       --output-dir-template string   go text template for generating the output directory (default "{{ .OutputDir }}{{ if .Release.TillerNamespace }}/{{ .Release.TillerNamespace }}{{ end }}{{ if .Release.Namespace }}/{{ .Release.Namespace }}{{ end }}{{ if .Release.KubeContext }}/{{ .Release.KubeContext }}{{ end }}/{{ .Release.Name }}/{{ .ChartName }}/{{ if .Release.Version }}{{ .Release.Version }}{{ else }}latest{{ end }}")
>       --skip-deps                    skip running "helm repo update" and "helm dependency build"
> ```

I am not sure where to put the default. I think a good place might be `NewFetchOptions()` in `pkg/config/fetch.go` but then I create an import cycle when I want to use the default in the tests I wrote in `pkg/state/state_test.go`.

Reply:
Thank you @elchenberg, for now we can leave the "long default one", at least we could harmonize the path generation with the GenerateChartPath() ;-)